sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -3,18 +3,17 @@
 
 package com.azure.core.util;
 
-import com.azure.core.util.logging.ClientLogger;
-
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 
 /**
  * A builder class that is used to create URLs.
  */
 public final class UrlBuilder {
-    private final ClientLogger logger = new ClientLogger(UrlBuilder.class);
+    private static final Map<String, UrlBuilder> PARSED_URLS = new HashMap<>();
```

Comment:
There might be a potential concurrency issue here on the static Map.

We had a possible one on this line in `UrlBuilder.parse`
```
PARSED_URLS.put(url, ub);
```

https://user-images.githubusercontent.com/15213781/92462976-c574af80-f1fd-11ea-8f19-33e0919913a5.png